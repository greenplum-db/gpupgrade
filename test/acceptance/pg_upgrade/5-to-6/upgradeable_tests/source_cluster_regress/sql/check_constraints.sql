-- Copyright (c) 2017-2023 VMware, Inc. or its affiliates
-- SPDX-License-Identifier: Apache-2.0

-- Test to ensure that tables with simple check constraints can be upgraded.

--------------------------------------------------------------------------------
-- Create and setup upgradeable objects
--------------------------------------------------------------------------------
CREATE TABLE heap_table_with_check_constraint (
    id INT,
    name text CHECK (id>=1 AND id<2)
);

INSERT INTO heap_table_with_check_constraint VALUES (1, 'Joe');
-- this insert should fail
INSERT INTO heap_table_with_check_constraint VALUES (2, 'Jane');

CREATE TABLE partition_table_with_check_constraint (
    a INT CONSTRAINT a_check CHECK (a+b>c),
    b INT,
    c INT) DISTRIBUTED BY (a)
    PARTITION BY RANGE(b)
        (PARTITION part START(0) END(4242));

INSERT INTO partition_table_with_check_constraint SELECT i, i, i FROM generate_series(1, 10) i;
-- this insert should fail
INSERT INTO partition_table_with_check_constraint VALUES (1, 1, 3);
