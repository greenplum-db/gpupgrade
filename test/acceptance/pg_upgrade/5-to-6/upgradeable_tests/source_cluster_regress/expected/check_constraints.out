-- Copyright (c) 2017-2023 VMware, Inc. or its affiliates
-- SPDX-License-Identifier: Apache-2.0

-- Test to ensure that tables with simple check constraints can be upgraded.

--------------------------------------------------------------------------------
-- Create and setup upgradeable objects
--------------------------------------------------------------------------------
CREATE TABLE heap_table_with_check_constraint ( id INT, name text CHECK (id>=1 AND id<2) );
CREATE

INSERT INTO heap_table_with_check_constraint VALUES (1, 'Joe');
INSERT 1
-- this insert should fail
INSERT INTO heap_table_with_check_constraint VALUES (2, 'Jane');
ERROR:  new row for relation "heap_table_with_check_constraint" violates check constraint "heap_table_with_check_constraint_id_check"

CREATE TABLE partition_table_with_check_constraint ( a INT CONSTRAINT a_check CHECK (a+b>c), b INT, c INT) DISTRIBUTED BY (a) PARTITION BY RANGE(b) (PARTITION part START(0) END(4242));
CREATE

INSERT INTO partition_table_with_check_constraint SELECT i, i, i FROM generate_series(1, 10) i;
INSERT 10
-- this insert should fail
INSERT INTO partition_table_with_check_constraint VALUES (1, 1, 3);
ERROR:  new row for relation "partition_table_with_check_constraint_1_prt_part" violates check constraint "a_check"
